runtime/common/BaseRestRemoteClient.h

```diff
@@ -323,6 +324,17 @@ class BaseRemoteRestRuntimeClient : public cudaq::RemoteRuntimeClient {
       if (optionalErrorMsg)
         *optionalErrorMsg = e.what();
       return false;
+    } catch (...) {
```

Comment:
Curious: what is going on here? Is the switch to `libc++` causing exceptions to be thrown that were not before? Seems like some comments are needed here. It's not entirely clear where the exception is (may be) coming from and why.

Reply:
This is what I had brought up in the standup at one point (also mentioned in the docs regarding integrating with third party libraries): I configured the static build of the libc++ to include everything it needs and hide its symbol, so that a CUDA-Q library can be created that can be later linked to other code built with a different compiler. However, this means that the exceptions thrown are not necessarily recognizable anymore by their type, since in that case the two libraries compiled with different libc++/libstdc++ will have their own definitions of the exception type. A catch all will still reliably catch them. There is one comment to that effect in the code as well, but I didn't add the comment everywhere that I made this change.

Reply:
Thank you for documenting the reasoning here. I remember that now.